src/plonk/circuit.rs

```diff
@@ -34,6 +37,32 @@ impl<C: ColumnType> Column<C> {
     }
 }
 
+impl<C: ColumnType> Ord for Column<C> {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        // This ordering is consensus-critical! The layouters rely on deterministic column
```

Comment:
Wish we had a different phrase than "consensus critical" because this is halo2, there might not be any consensus involved in a downstream application. We need a word for "if the behavior of this changes for the prover, the verifier may reject the proof." Deterministic isn't enough.

Reply:
"Circuit-critical"? We can figure this out later.

Reply:
Maybe something like 'semantically binding'?